boot/boot_serial/src/boot_serial.c

```diff
@@ -439,6 +439,39 @@ bs_upload(char *buf, int len)
     flash_area_close(fap);
 }
 
+#ifdef MCUBOOT_BOOT_MGMT_ECHO
+static bool 
+decode_echo(cbor_state_t *state, cbor_string_type_t *result)
+{
+    size_t bsstrdecoded;
+
+    if (!map_start_decode(state)) {
+        return false;
+    }
+    if (!multi_decode(2, 2, &bsstrdecoded, (void *)tstrx_decode, state, result, sizeof(cbor_string_type_t))) {
+        return false;
+    }
+    map_end_decode(state);
+    return true;
```

Comment:
Shouldn't the map_end_decode happen regardless whether multi_decode is successful or not?
I have checked decode_Upload and it seems to call on map_end_decode independently from `multi_decode` returned value.

Reply:
It's a pain to read it, but you're right, it seems like doing so. I've added the end decode now